Optional Chaining: Add intros to currently context-less examples (#15352)
mdn/content: files/en-us/web/javascript/reference/operators/optional_chaining/index.md

### Optional chaining not valid on the left-hand side of an assignment

> It is invalid to try to assign to the result of a optional chaining expression:

Reviewer: ```suggestion
It is invalid to try to assign to the result of an optional chaining expression:
```

```js
let object = {};
object?.property = 1; // Uncaught SyntaxError: Invalid left-hand side in assignment
```

### Array item access with optional chaining

> You can use the [the bracket notation](/en-US/docs/Web/JavaScript/Reference/Operators/Property_Accessors#bracket_notation) to use the optional chaining on arrays:

Reviewer: ```suggestion
You can use [bracket notation](/en-US/docs/Web/JavaScript/Reference/Operators/Property_Accessors#bracket_notation) for optional chaining on arrays:
```
